packages/amplify_auth_plugin_interface/lib/src/SignUp/ResendSignUpCodeResult.dart

```diff
@@ -19,7 +19,7 @@ class ResendSignUpCodeResult {
   AuthCodeDeliveryDetails codeDeliveryDetails;
   ResendSignUpCodeResult({required codeDeliveryDetails})
       : this.codeDeliveryDetails = AuthCodeDeliveryDetails(
-            attributeName: codeDeliveryDetails["attributeName"],
-            deliveryMedium: codeDeliveryDetails["deliveryMedium"],
-            destination: codeDeliveryDetails["destination"]);
+            attributeName: codeDeliveryDetails["attributeName"] ?? "",
+            deliveryMedium: codeDeliveryDetails["deliveryMedium"] ?? "",
+            destination: codeDeliveryDetails["destination"] ?? "");
```

Comment:
Didn't we make a change to `AuthCodeDeliveryDetails` that would handle null parameters in the other null safety PR?

Reply:
Yes we made changes but it doesn't work.  The `AuthCodeDeliveryDetails` class has non nullable string fields that have a default value.  This does not mean that the parameters can be null though.  So because `codeDeliveryDetails["attributeName"]` is nullable, there is a compile error.  That is why I added `?? ""` in the end.  

These changes are based on the assumption that the String fields of `AuthCodeDeliveryDetails` cannot be non nullable but can be not initialized and have a value of "" as well.  @haverchuck do you have any additional insight here?

Reply:
Ah hm, I wonder if we made a mistake with the `AuthCodeDeliveryDetails` class then. I'm thinking if it was:
```dart
class AuthCodeDeliveryDetails {
  String? attributeName;
  
  AuthCodeDeliveryDetails({ this. attributeName = "" });
}
```

Then something like this should have no issues:
```dart
String? somethingNullable;
AuthCodeDeliveryDetails deets = AuthCodeDeliveryDetails(attributeName: somethingNullable);
```

But if that was a miss during the core migration, we'll just have to live with it for now and move on. Maybe come back to it later since, in the end, it's really not blocker in any way.

Reply:
> The second suggestion of making the class's field optional (and a default "") is equally inaccurate

I would agree that this would be generally inaccurate in a vacuum. But the suggestion was made with consideration that existing code was already doing something similar in another PR and I did not want to infer too deeply what the original author's intent was. We 100% do not want to be masking compiler errors so maybe it's worth diving deeper into the class but I could not rule out that `""` was intentionally acceptable.

Reply:
@haverchuck could we have your thoughts on your intention there?  Looking more at the code, I'm guessing these should be nullable fields and that the default "" was added before to allow for that in some way.

Reply:
The empty string is definitely the wrong thing here - my mistake.  Making the individual values null would be an improvement, although the truly correct thing would be to make the AuthCodeDeliveryDetails value itself nullable. This might require some changes at the platform level.  If you have bandwidth this would be the right change in terms of library consistency.  Sorry for creating this issue in the first place!

Reply:
Sounds good, updated AuthCodeDeliveryDetails to have nullable fields: 

```
class AuthCodeDeliveryDetails {
  String? attributeName;
  String? deliveryMedium;
  String? destination;
```

Reply:
Good catch, thanks